dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -317,28 +318,32 @@ private WorkflowVersion updateDBVersionSourceFilesWithRemoteVersionSourceFiles(W
     protected List<Workflow> githubWebhookRelease(String repository, String username, String gitReference, String installationId) {
         // Retrieve the user who triggered the call (must exist on Dockstore if workflow is not already present)
         User user = GitHubHelper.findUserByGitHubUsername(this.tokenDAO, this.userDAO, username, false);
-
-        // Get Installation Access Token
-        String installationAccessToken = gitHubAppSetup(installationId);
-
         // Grab Dockstore YML from GitHub
-        GitHubSourceCodeRepo gitHubSourceCodeRepo = (GitHubSourceCodeRepo)SourceCodeRepoFactory.createGitHubAppRepo(installationAccessToken);
+        GitHubSourceCodeRepo gitHubSourceCodeRepo = (GitHubSourceCodeRepo)SourceCodeRepoFactory.createGitHubAppRepo(gitHubAppSetup(installationId));
 
+        GHRateLimit startRateLimit = gitHubSourceCodeRepo.getGhRateLimitQuietly();
+        GHRateLimit endRateLimit;
+        boolean isSuccessful = false;
         try {
+
             SourceFile dockstoreYml = gitHubSourceCodeRepo.getDockstoreYml(repository, gitReference);
             // If this method doesn't throw an exception, it's a valid .dockstore.yml with at least one workflow or service.
             // It also converts a .dockstore.yml 1.1 file to a 1.2 object, if necessary.
             final DockstoreYaml12 dockstoreYaml12 = DockstoreYamlHelper.readAsDockstoreYaml12(dockstoreYml.getContent());
             final List<Workflow> workflows = new ArrayList();
             workflows.addAll(createServicesAndVersionsFromDockstoreYml(dockstoreYaml12.getService(), repository, gitReference,
-                    gitHubSourceCodeRepo, user, dockstoreYml));
+                    (GitHubSourceCodeRepo)SourceCodeRepoFactory.createGitHubAppRepo(gitHubAppSetup(installationId)), user, dockstoreYml));
```

Comment:
I gave you and asked you to put in this code, but my code was wrong. :( I wanted a new token to be created for each workflow/service, but looking at it, it's not really going to save much -- you'll get a slightly newer token after it's read the .dockstore.yml, but if the loop needs to process 15 workflows, it will still be using the same token for the first one and the 15th.

You should pass installationId as the first parameter, and then in `createServicesAndVersionsFromDockstoreYml`call `SourceCodeRepoFactory.createGitHubAppRepo`